server/user_account/tests.py

```diff
@@ -79,19 +79,19 @@ def test_registration_success(self):
                 "last_name": "user",
                 "role": "SA",
                 "is_active": "True"}
-        response = self.client.post("/registration/", data)
+        response = self.client.post("/open-registration/", data)
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
     def test_registration_failure(self):
         """ User can't register if missing fields """
         data = {'user_name': 'test_case', "password": "password", "first_name": "test",
                 "last_name": "user", "role": "SA"}
-        response = self.client.post("/registration/", data)
+        response = self.client.post("/open-registration/", data)
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
     def test_registration_unauthorized_request(self):
         """ User can't access the GET method at this particular endpoint """
-        response = self.client.get("/registration/")
+        response = self.client.get("/open-registration/")
         self.assertEqual(response.status_code,
                          status.HTTP_405_METHOD_NOT_ALLOWED)
 
```

Comment:
Can you please add a test case for the new registration view you implemented. Also don't forget to test for users who don't have a token when doing the request.

Reply:
I added a detailed test case 5246f229da4dca5117c5e85036a64cd0d41ecd18
Only thing missing is handling the rules for who can register who. Once that is implemented, more test cases can be added.